Kernel/Bus/USB/EHCI/Registers.h

```diff
@@ -0,0 +1,284 @@
+    // Note: Only up to 14 bits are actually used, and the last 3 bits must never be `000` or `111`
+    volatile u32 frame_index;
+
+    // 2.3.5 CTRLDSSEGMENT - Control Data Structure Segment Register
+    // Note: We use `volatile` to ensure 32 bit writes
+    // Note: Upper 32 bits of periodic-frame- and asynchronous-list pointers
+    volatile u32 segment_selector;
+
+    // 2.3.6 PERIODICLISTBASE - Periodic Frame List Base Address Register
+    // Note: We use `volatile` to ensure 32 bit writes
+    // Note: Page-aligned addresses only
+    volatile u32 frame_list_base_address;
+    // 2.3.7 ASYNCLISTADDR - Current Asynchronous List Address Register
+    // Note: We use `volatile` to ensure 32 bit writes
+    // Note: 32 byte (cache-line) aligned addresses only
+    volatile u32 next_asynchronous_list_address;
+
+    u32 _padding[9];
+
+    // 2.3.8 CONFIGFLAG - Configure Flag Register
+    u32 configured_flag;
+
+    union PortStatusControl {
+        enum class LineStatus : u32 {
+            SE0 = 0b00,
+            J_State = 0b10,
+            K_State = 0b01,
+            Undefined = 0b11
+        };
+        enum class PortIndicatorControl : u32 {
+            Off = 0b00,
+            Amber = 0b01,
+            Green = 0b10,
+            Undefined = 0b11
+        };
+        enum class PortTestControl : u32 {
+            NotEnabled = 0b0000,
+            J_State = 0b0001,
+            K_State = 0b0010,
+            SE0_NAK = 0b0011,
+            Packet = 0b0100,
+            Force_Enable = 0b0101,
+        };
+
+        const struct {
+            u32 current_connect_status : 1;
+            u32 connect_status_change : 1; // R/WC
+            u32 port_enable : 1;
+            u32 port_enable_change : 1; // R/WC
+            u32 over_current_active : 1;
+            u32 over_current_change : 1; // R/WC
+            u32 force_resume : 1;
+            u32 suspend : 1;
+            u32 port_reset : 1;
+            u32 : 1;
+            LineStatus line_status : 2;
+            u32 port_power : 1;
+            u32 port_owner : 1;
+            PortIndicatorControl port_indicator_control : 2;
+            PortTestControl port_test_control : 4;
+            u32 wake_on_connect_enable : 1;      // WKCNNT_E
+            u32 wake_on_disconnect_enable : 1;   // WKDSCNNT_E
+            u32 wake_on_over_current_enable : 1; // WKOC_E
+            u32 : 9;
+        };
+        u32 raw;
+    } port_status_control[];
+    static_assert(AssertSize<PortStatusControl, 32 / 8>());
+};
+// Table 2-8. Host Controller Operational Registers
+static_assert(__builtin_offsetof(OperationalRegisters, command) == 0x00);
+static_assert(__builtin_offsetof(OperationalRegisters, status) == 0x04);
+static_assert(__builtin_offsetof(OperationalRegisters, interrupt_enable) == 0x08);
+static_assert(__builtin_offsetof(OperationalRegisters, frame_index) == 0x0C);
+static_assert(__builtin_offsetof(OperationalRegisters, segment_selector) == 0x10);
```

Comment:
I wonder if we should start using that OFFSET_OF macro from AK/Types.h for new code. food for thought, not for changes though :)